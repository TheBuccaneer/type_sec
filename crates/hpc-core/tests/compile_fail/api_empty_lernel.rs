use hpc_core::*;

fn main() {
    let ctx = Context::create_context().unwrap();
    let queue = ctx.create_queue().unwrap();

    // Buffer ist noch Empty
    let buf = ctx.create_buffer::<u8>(16).unwrap();

    // Kernel starten mit Empty ( illegal)
    let (_inflight, _evt) = buf.enqueue_kernel(&queue, &unimplemented!(), 16).unwrap();
}
