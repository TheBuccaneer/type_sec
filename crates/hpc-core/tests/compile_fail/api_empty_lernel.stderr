error[E0599]: no method named `enqueue_kernel` found for struct `DeviceBuffer<'_, u8, hpc_core::Empty>` in the current scope
  --> tests/compile_fail/api_empty_lernel.rs:11:33
   |
11 |     let (_inflight, _evt) = buf.enqueue_kernel(&queue, &unimplemented!(), 16).unwrap();
   |                                 ^^^^^^^^^^^^^^ method not found in `DeviceBuffer<'_, u8, hpc_core::Empty>`
   |
   = note: the method was found for
           - `DeviceBuffer<'brand, T, hpc_core::Ready>`

warning: unreachable expression
  --> tests/compile_fail/api_empty_lernel.rs:11:56
   |
11 |     let (_inflight, _evt) = buf.enqueue_kernel(&queue, &unimplemented!(), 16).unwrap();
   |                                                        ^----------------
   |                                                        ||
   |                                                        |any code following this expression is unreachable
   |                                                        unreachable expression
   |
   = note: `#[warn(unreachable_code)]` on by default
